diff --git a/kernel/apk_sign.c b/kernel/apk_sign.c
index 3f2ae2d4..96a3c861 100644
--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -18,6 +18,10 @@
 #include "app_profile.h"
 #include "klog.h" // IWYU pragma: keep
 
+#ifdef KSU_MANAGER_PACKAGE
+#define WILD_KSU_PACKAGE_NAME "com.twj.wksu"
+#endif
+
 struct sdesc {
 	struct shash_desc shash;
 	char ctx[];
@@ -356,9 +360,11 @@ bool is_manager_apk(char *path)
 	}
 
 	// pkg is `<real package>`
-	if (strncmp(pkg, KSU_MANAGER_PACKAGE, sizeof(KSU_MANAGER_PACKAGE))) {
+	if (strncmp(pkg, WILD_KSU_PACKAGE_NAME, sizeof(WILD_KSU_PACKAGE_NAME))) {
 		return false;
 	}
 #endif
-	return check_v2_signature(path, EXPECTED_MANAGER_SIZE, EXPECTED_MANAGER_HASH);
+	return check_v2_signature(path, 0x38a, "670a2656d6c353487e5c965c1b3a6074077ce7562b03cc5254e28905dfd93d80");
 }
